care/facility/api/viewsets/icd.py

```diff
@@ -11,11 +11,25 @@ class ICDViewSet(ViewSet):
     def serialize_data(self, objects: list[ICD11]):
         return [diagnosis.get_representation() for diagnosis in objects]
 
-    def retrieve(self, request, pk):
-        obj = get_icd11_diagnosis_object_by_id(pk, as_dict=True)
-        if not obj:
-            raise Http404
-        return Response(obj)
+    def retrieve(self, request, pk: int):
+        try:
+            pk = int(pk)
+        except ValueError as err:
+            raise ValidationError(detail="ID must be an integer.") from err
+
+        diagnosis, error = get_icd11_diagnosis_object_by_id(pk, as_dict=True)
```

Comment:
@sainak should we switch to raising exceptions instead of returning tuple containing error?

Reply:
yeah lets follow python way

Reply:
I was facing problem with raising exceptions directly from the function 

1. unable to provide a detailed error in api responses.
2. For above issue we would need to raise APIExceptions from the function but since that is also used at other non views functions, we shouldn't use APIExceptions in the function

Reply:
you'll have to extend the apiexception to give the specific method, that will also allow in better error handelling